gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMessageHttpRequestFormatter.java

```diff
@@ -56,20 +65,17 @@
   }
 
   @Override
-  public Map<String, String> getPathParams(ApiMessage apiMessage, Set<String> paramNames) {
-    Map<String, String> pathParams = new HashMap<>();
-    Map<String, List<String>> pathParamMap = apiMessage.populateFieldsInMap(paramNames);
-    Iterator<Map.Entry<String, List<String>>> iterator = pathParamMap.entrySet().iterator();
-    while (iterator.hasNext()) {
-      Map.Entry<String, List<String>> pair = iterator.next();
-      pathParams.put(pair.getKey(), pair.getValue().get(0));
-    }
-    return pathParams;
+  public Map<String, String> getPathParams(T apiMessage, String resourceNameField) {
+    Map<String, List<String>> pathParamMap =
+        apiMessage.populateFieldsInMap(Collections.singleton(resourceNameField));
+
+    String resourceNamePath = pathParamMap.get(resourceNameField).get(0);
```

Comment:
If the resource name field wasn't found, we should throw an exception

Reply:
Done.